### PR TITLE
Feature/#135 docker

### LINKED ISSUE #135: Deployment & Release 1.1.0
We need a deployment with all the changes that we've made in this sprint. This might include going from war to jar which in turn means setting up a docker image.

### LINKED ISSUE #151: Automated deployment
I can set up automated deployment using semaphore, circle ci or travis. I'd prefer doing it with Semaphore because I can easily add configuration files that will be present during the deployment. In addition, Semaphore is tamper-proof since one can't just change a file in the repository and potentially execute commands on the server side.
I can do it with all of them though. You guys choose @outcobra/developers 

This depends on #68 #140 and fixes #135

### PR DESCRIPTION
**Pull request by jmesserli**
This closes #135 and closes #151

Summary of changes:
- ​Switched from war to jar packaging
- Added git-info plugin for spring-boot-actuator (try here: https://outcobra-a.pegnu.cloud/backend/info)
- Created docker containers for backend and frontend, wrote docker-compose.yml with a mysql database which bundles all three
- Automated deployment using semaphore and pkg.pegnu.cloud as Docker registry

I made sure to:
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Follow the contribution guidelines
- [x] Update CI scripts on Semaphore (changes to `auth0.yml` or `application.yml`)

Please review @outcobra/developers.

### CODE REVIEW

File: backend/build.gradle [hunk 1]
```diff
@@ -24,14 +27,22 @@ apply plugin: 'kotlin'
 //apply plugin: 'kotlin-spring'
```
Comment (needToRoll) [inline backend/build.gradle]: why do you disable kotlin-spring? We need this... at least with kotlin v1.1.1

**Comment by jmesserli** [comment]
Kotlin 1.1 is not in this branch

needToRoll <notifications@github.com> schrieb am So., 19. März 2017, 08:54:

**Comment by jmesserli** [comment]
Of course not, this is just a compose file to check if a prod-like
deployment would work

needToRoll <notifications@github.com> schrieb am So., 19. März 2017, 08:58:

**Comment by jmesserli** [comment]
Please re-review

**Pull request merged**